sshd-common/src/main/java/org/apache/sshd/common/config/PreferredAuthConfigEntry.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sshd.common.config;
+
+import java.util.Objects;
+
+import org.apache.sshd.common.util.ValidateUtils;
+
+/**
+ * @author <a href="mailto:dev@mina.apache.org">Apache MINA SSHD Project</a>
+ */
+public final class PreferredAuthConfigEntry {
+    private final String configKeyName;
+    private final String authFactoryName;
+    private final boolean enabledByDefault;
+    private final int hashValue;
+    private final String strValue;
+
+    public PreferredAuthConfigEntry(
+                                    String configKeyName, String authFactoryName, boolean enabledByDefault) {
+        this.configKeyName = ValidateUtils.checkNotNullAndNotEmpty(configKeyName, "No configuration key name provided");
+        this.authFactoryName
+                = ValidateUtils.checkNotNullAndNotEmpty(authFactoryName, "No authentication factory name provided");
+        this.enabledByDefault = enabledByDefault;
+        this.hashValue = Objects.hash(configKeyName, authFactoryName, enabledByDefault);
+        this.strValue
```

Comment:
Does it really make sense to pre-compute this toString() value? Normally I would expect toString() to be called only for debugging output.

Reply:
True, but since all members are *final* I think it makes sense

Reply:
It wastes space, even if toString() is never called.

Reply:
I'll consider it